### PR TITLE
core: GoogleAIStudio support for tool use

### PR DESCRIPTION
**Pull request by spolu**
## Description

- Add support for getting errors from GoogleAIStudio
- Add support for tool use in GoogleAIStudio

## Risk

Low.

## Deploy Plan

- deploy `core`

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
Nice 👍  

LGTM assuming this was tested

File: core/src/providers/google_ai_studio.rs
Comment (flvndvd) [inline core/src/providers/google_ai_studio.rs:553]: 😢

File: core/src/providers/google_ai_studio.rs
Comment (flvndvd) [inline core/src/providers/google_ai_studio.rs:538]: Not sure to get the value of using a for loop here 🙃

File: core/src/providers/google_ai_studio.rs
Comment (flvndvd) [inline core/src/providers/google_ai_studio.rs:679]: ✂️

File: core/src/providers/google_ai_studio.rs
Comment (flvndvd) [inline core/src/providers/google_ai_studio.rs:705]: ✂️

File: core/src/providers/google_ai_studio.rs
Comment (flvndvd) [inline core/src/providers/google_ai_studio.rs:895]: ✂️

**Review by flvndvd — COMMENTED** [review]
\o/

File: core/src/providers/google_ai_studio.rs
Comment (spolu) [reply core/src/providers/google_ai_studio.rs:555]: simpler error reporting. Otherwise I need to return a Result and as a consequence use a map etc...

Comment (fontanierh) [reply core/src/providers/google_ai_studio.rs:555]: Would probably be more readable, more idiomatic, and shorter though

**Comment by spolu** [comment]
Blocked on https://github.com/dust-tt/dust/pull/5497

**Pull request merged**
